pkg/controller/controllercmd/builder.go

```diff
@@ -58,6 +58,8 @@ func NewController(componentName string, startFunc StartFunc) *ControllerBuilder
 	}
 }
 
+// WithRestartOnChange will enable a file observer controller loop that observes changes into specified files. If a change to file is detected,
+// the operator container will restart to reload the resources.
```

Comment:
better: the process will be terminated with return code zero. The restart is not done by this logic, but by the pod around the binary.